### PR TITLE
Add RBAC config and readme blurb

### PR DESCRIPTION
**Pull request by rmars**
If RBAC is enabled, we need to explicitly allow linkerd access to services/endpoints. Add a RBAC config for this.

Tested on Kevin's 1.7 cluster.

### CODE REVIEW

**Comment by esbie** [comment]
Not that it's much more prominent, but this should be in https://discourse.linkerd.io/t/flavors-of-kubernetes/53

File: k8s-daemonset/k8s/linkerd-rbac-beta.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: endpoints-reader
+rules:
+  - apiGroups: [""] # "" indicates the core API group
+    resources: ["endpoints", "services"]
+    verbs: ["get", "watch", "list"]
+  - apiGroups: ["l5d.io"]
```
Comment (rmars) [inline k8s-daemonset/k8s/linkerd-rbac-beta.yaml]: I think I'll split this into another Role for clarity

**Comment by rmars** [comment]
@esbie yeah good idea, I'll update that with the readme text and a link to linkerd-rbac-beta.yml once this is merged.

File: k8s-daemonset/k8s/linkerd-rbac-beta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# RBAC configs for linkerd
+---
+# allow linkerd/namerd permissions to enable service discovery
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: endpoints-reader
+rules:
+  - apiGroups: [""] # "" indicates the core API group
+    resources: ["endpoints", "services"]
```
Comment (rmars) [inline k8s-daemonset/k8s/linkerd-rbac-beta.yaml]: yeahhh I see. I think I'll add pods and then also add a comment explaining that pods are only required for "*-legacy.yml" configs :) That way this rbac config will work for all the examples, but people who want to be more restrictive can modify this.

File: k8s-daemonset/k8s/linkerd-rbac-beta.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# RBAC configs for linkerd
+---
+# allow linkerd/namerd permissions to enable service discovery
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: endpoints-reader
```
Comment (rmars) [inline k8s-daemonset/k8s/linkerd-rbac-beta.yaml]: good idea! thanks!

File: k8s-daemonset/README.md [hunk 4]
```diff
@@ -358,3 +358,65 @@ Add the `-streaming` flag to send a streaming gRPC request:
...
+service discovery).
+
+You can find out if your cluster supports RBAC by running:
+```
+kubectl api-versions | grep rbac
+```
+
+If you do, you'll need to grant linkerd/namerd access:
+```
+kubectl apply -f k8s/linkerd-rbac-beta.yaml
```
Comment (klingerf) [inline k8s-daemonset/README.md]: This should be `k8s/jenkins-rbac-beta.yml`

Comment (rmars) [inline k8s-daemonset/README.md]: eek, nice catch!

**Pull request merged**